addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -75,15 +96,216 @@ public void importFromUrl(final String url) throws IllegalArgumentException, IOE
     public void importCollection(String collection) throws JsonProcessingException {
         PostmanCollection postmanCollection = parse(collection);
 
-        // TODO: Extract list of HttpMessage from PostmanCollection and send requests
+        List<HttpMessage> httpMessages = new ArrayList<>();
+        extractHttpMessages(postmanCollection.getItem(), httpMessages);
+
+        requestor.run(httpMessages);
     }
 
     public PostmanCollection parse(String collectionJson) throws JsonProcessingException {
         ObjectMapper objectMapper = new ObjectMapper();
         return objectMapper.readValue(collectionJson, PostmanCollection.class);
     }
 
+    private void extractHttpMessages(List<AbstractItem> items, List<HttpMessage> httpMessages) {
+        for (AbstractItem item : items) {
+            if (item instanceof Item) {
+                HttpMessage httpMessage = extractHttpMessage((Item) item);
+                if (httpMessage != null) {
+                    httpMessages.add(httpMessage);
+                }
+            } else if (item instanceof ItemGroup) {
+                extractHttpMessages(((ItemGroup) item).getItem(), httpMessages);
+            }
+        }
+    }
+
     private static boolean isSupportedScheme(String scheme) {
         return "http".equalsIgnoreCase(scheme) || "https".equalsIgnoreCase(scheme);
     }
+
+    public static boolean isContentTypeAlreadySet(List<KeyValueData> headers) {
+        for (KeyValueData header : headers) {
+            if (header.getKey().equalsIgnoreCase("Content-Type")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static HttpMessage extractHttpMessage(Item item) {
+        Request request = item.getRequest();
+        Url url = request.getUrl();
```

Comment:
If `request` can be `null` this needs to be done after the if. Although request is a required filed, should we fail/warn instead? We should talk about error reporting.
Tests are needed to cover the different representations (e.g. request as object and as string, same for other properties).

Reply:
Request is required, will remove the null check.

Reply:
But we should still validate that we have one, the user can try import an invalid collection. Regarding the error reporting, I think that can be done in a follow up PR.

Reply:
> But we should still validate that we have one, the user can try import an invalid collection.

I believe [this](https://github.com/zaproxy/zap-extensions/blob/ed8f0469cd7ed4bfcafda99849736ab138942861/addOns/postman/src/test/java/org/zaproxy/addon/postman/ListDeserializerUnitTest.java#L62) covers it, no?

Reply:
That confirms it's able to parse without the `request` but not that it's able to import it (for that it would have to call `importCollection(String)`).